"panic: no expansion registered" during import (modules not expanded)

### Terraform Version

```
Terraform v0.13.0-dev - master from today
+ provider registry.terraform.io/-/aws v2.65.0
+ provider registry.terraform.io/hashicorp/aws v2.65.0
```

This appears the same panic output as https://github.com/hashicorp/terraform/pull/25176 - But I built from master just now and still getting the error.

To note: Everytime I run it, I get a different module name in the panic - ie,
panic: no expansion has been registered for module.s3
panic: no expansion has been registered for module.rds
panic: no expansion has been registered for module.networking
which just cycles thru the modules

### Terraform Configuration Files

Running on remote backend to TFC

```hcl
main
# main.tf
# addition for TF Cloud Migration
terraform {
  backend "remote" {
    hostname     = "app.terraform.io"
    organization = "pilotsnpaws"

    workspaces {
      name = "aws-infras-build"
    }
  }
}

# Setup AWS, using Ohio for now
# Requires aws configure to be run locally to setup AWS ID and Key
provider "aws" {
  version = "~> 2.54"
  profile = "pilotsnpaws"
  region  = var.aws_region
}

module "s3" {
  source = "./modules/s3"
}

module "rds" {
  source                = "./modules/rds"
  rds_security_group_id = module.networking.rds_security_group_id
}

module "compute" {
  source = "./modules/compute"
}

# Even though we use networking, via the webservers module, to create security groups, 
# having it here is redundant to calling it in the compute module and will result in 
# duplicate sec groups being planned 

module "networking" {
  source = "./modules/networking"
}


# aws-infras-build/modules/compute.hcl
# Create instances, how many defined in var instance_count

data "aws_availability_zones" "available" {
  state = "available"
}

resource "aws_instance" "dev-wordpress" {

}



```

### Debug Output
https://gist.github.com/mikegreen/818c00744c793289ba0acd76ec33e4ee

### Crash Output
Console output:
panic: no expansion has been registered for module.rds

Crash.log - https://gist.github.com/mikegreen/0c428725c17bab897e01e1a78eb36bbd


### Expected Behavior
TF would Import resource (or at least try, my syntax could be wrong)

### Actual Behavior
Import crash/panic

### Steps to Reproduce

1. Init works fine
1. terraform import module.compute.aws_instance.dev-wordpress i-097f4adc96413b96a
1. This leaves workspace in locked state
1. terraform force-unlock -force pilotsnpaws/aws-infras-build
1. Trying again, same panic except it will iterate thru the modules defined

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

### References
https://github.com/hashicorp/terraform/pull/25176

--- a/terraform/context_import_test.go
+++ b/terraform/context_import_test.go
@@ -527,7 +527,7 @@ func TestContextImport_module(t *testing.T) {
 	state, diags := ctx.Import(&ImportOpts{
 		Targets: []*ImportTarget{
 			&ImportTarget{
-				Addr: addrs.RootModuleInstance.Child("child", addrs.NoKey).ResourceInstance(
+				Addr: addrs.RootModuleInstance.Child("child", addrs.IntKey(0)).ResourceInstance(
 					addrs.ManagedResourceMode, "aws_instance", "foo", addrs.NoKey,
 				),
 				ID: "bar",
@@ -565,7 +565,7 @@ func TestContextImport_moduleDepth2(t *testing.T) {
 	state, diags := ctx.Import(&ImportOpts{
 		Targets: []*ImportTarget{
 			&ImportTarget{
-				Addr: addrs.RootModuleInstance.Child("child", addrs.NoKey).Child("nested", addrs.NoKey).ResourceInstance(
+				Addr: addrs.RootModuleInstance.Child("child", addrs.IntKey(0)).Child("nested", addrs.NoKey).ResourceInstance(
 					addrs.ManagedResourceMode, "aws_instance", "foo", addrs.NoKey,
 				),
 				ID: "baz",
@@ -603,7 +603,7 @@ func TestContextImport_moduleDiff(t *testing.T) {
 	state, diags := ctx.Import(&ImportOpts{
 		Targets: []*ImportTarget{
 			&ImportTarget{
-				Addr: addrs.RootModuleInstance.Child("child", addrs.NoKey).ResourceInstance(
+				Addr: addrs.RootModuleInstance.Child("child", addrs.IntKey(0)).ResourceInstance(
 					addrs.ManagedResourceMode, "aws_instance", "foo", addrs.NoKey,
 				),
 				ID: "baz",
@@ -765,7 +765,7 @@ aws_instance.foo.0:
 
 const testImportModuleStr = `
 <no state>
-module.child:
+module.child[0]:
   aws_instance.foo:
     ID = foo
     provider = provider["registry.terraform.io/hashicorp/aws"]
@@ -773,7 +773,7 @@ module.child:
 
 const testImportModuleDepth2Str = `
 <no state>
-module.child.nested:
+module.child[0].nested:
   aws_instance.foo:
     ID = foo
     provider = provider["registry.terraform.io/hashicorp/aws"]

--- a/terraform/graph_builder_import.go
+++ b/terraform/graph_builder_import.go
@@ -79,9 +79,20 @@ func (b *ImportGraphBuilder) Steps() []GraphTransformer {
 		// analyze the configuration to find references.
 		&AttachSchemaTransformer{Schemas: b.Schemas, Config: b.Config},
 
+		// Create expansion nodes for all of the module calls. This must
+		// come after all other transformers that create nodes representing
+		// objects that can belong to modules.
+		&ModuleExpansionTransformer{
+			Config: b.Config,
+		},
+
 		// Connect so that the references are ready for targeting. We'll
 		// have to connect again later for providers and so on.
 		&ReferenceTransformer{},
+
+		// Make sure data sources are aware of any depends_on from the
+		// configuration
+		&attachDataResourceDependenciesTransformer{},
 
 		// This validates that the providers only depend on variables
 		&ImportProviderValidateTransformer{},

--- a/terraform/testdata/import-module/main.tf
+++ b/terraform/testdata/import-module/main.tf
@@ -3,6 +3,7 @@ provider "aws" {
 }
 
 module "child" {
+  count = 1
   source = "./child"
   providers = {
     aws = aws

